features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -79,15 +61,10 @@ export default function Environments() {
     const [selectedMgLabel, setSelectedMgLabel] = useState([...api.labels]);
 
     const [isUpdating, setUpdating] = useState(false);
-
-    const restApi = new API();
-    const [mgLabels, setMgLabels] = useState([]);
-    useEffect(() => {
-        restApi.microgatewayLabelsGet()
-            .then((result) => {
-                setMgLabels(result.body.list);
-            });
-    }, []);
+    let isWebsocket = false;
+    if (api) {
+        isWebsocket = (api.type === 'WS');
+    }
```

Comment:
Shall we move this logic to [`Resource` ](https://github.com/wso2/carbon-apimgt/blob/master/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/Resource.js#L49-L51) abstraction level. So that 'WS' is not getting hardcoded everywhere

Reply:
Done